File: mpp-proiect-csharp-culbec/Server/Service/ServiceRaceImpl.cs
using Common.Exceptions;
using log4net;
using Model;
using Persistence.Repository;

namespace Server.Service;

public class ServiceRaceImpl : IServiceRace
{
    private static readonly ILog Log = LogManager.GetLogger(typeof(ServiceRaceImpl));
    private readonly IRepositoryRace _repositoryRace;

    public ServiceRaceImpl(IRepositoryRace repositoryRace)
    {
        Log.Info("Initializing ServiceRaceImpl.");
        _repositoryRace = repositoryRace;
        Log.Info("ServiceRaceImpl initialized.");
    }

    public IEnumerable<Race> FindAllRaces()
    {
        Log.Info("Finding all the races stored in the repository...");

        try
        {
            var races = _repositoryRace.FindAll()!.ToList();

            Log.InfoFormat("Found {0} races in the repository", races.Count);
            return races;
        }
        catch (RepositoryException e)
        {
            Log.ErrorFormat("Something went wrong while retrieving all the races: {0}", e.Message);
            throw new ServiceException("Something went wrong while retrieving all the races: " + e.Message);
        }
    }

    public IEnumerable<int> FindAllRaceEngineCapacities()
    {
        Log.Info("Finding all the race engine capacities...");

        try
        {
            var engineCapacities = _repositoryRace.FindAllRaceEngineCapacities().ToList();

            Log.InfoFormat("Found {0} engine capacities in the repository", engineCapacities.Count);
            return engineCapacities;
        }
        catch (RepositoryException e)
        {
            Log.ErrorFormat("Something went wrong while retrieving all the engine capacities: {0}", e.Message);
            throw new ServiceException("Something went wrong while retrieving all the engine capacities: " + e.Message);
        }
    }
}

File: mpp-proiect-java-culbec/MotorcycleContest/Server/src/main/java/Service/ServiceRaceImpl.java
package Service;

import Repository.RepositoryException;
import Exceptions.ServiceException;
import Model.Race;
import Repository.RaceRepository;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.util.List;

public class ServiceRaceImpl implements ServiceRace {
    private static final Logger logger = LogManager.getLogger();
    private final RaceRepository raceRepository;

    public ServiceRaceImpl(RaceRepository raceRepository) {
        logger.traceEntry("Initializing the ServiceRaceImpl with the RaceRepository...");
        this.raceRepository = raceRepository;
        logger.traceExit("Initialized the ServiceRaceImpl!");
    }

    @Override
    public Iterable<Race> findAllRaces() throws ServiceException {
        logger.traceEntry("Finding all the races...");

        try {
            List<Race> races = (List<Race>) this.raceRepository.findAll();

            logger.traceExit("Found all the races!");
            return races;
        } catch (RepositoryException e) {
            logger.error("Couldn't find all the races: {}", e.getMessage());
            throw new ServiceException("Couldn't find all the races: " + e.getMessage());
        }
    }

    @Override
    public Iterable<Integer> findAllRaceEngineCapacities() throws ServiceException {
        logger.traceEntry("Finding all the race engine capacities...");

        try {
            List<Integer> engineCapacities = (List<Integer>) this.raceRepository.findAllRaceEngineCapacities();

            logger.traceExit("Found all the engine capacities!");
            return engineCapacities;
        } catch (RepositoryException e) {
            logger.error("Couldn't find all the race engine capacities: {}", e.getMessage());
            throw new ServiceException("Couldn't find all the race engine capacities: " + e.getMessage());
        }
    }
}
